app/src/org/commcare/activities/FormEntryActivityUIController.java

```diff
@@ -716,6 +717,8 @@ protected void updateFormRelevancies() {
                 FormRelevancyUpdating.getOldSelectChoicesForEachWidget(oldWidgets);
         ArrayList<String> oldQuestionTexts =
                 FormRelevancyUpdating.getOldQuestionTextsForEachWidget(oldWidgets);
+        ArrayList<IAnswerData> oldAnswers =
```

Comment:
I'm really, really sorry for making this suggestion and then retracting it, but I'm almost positive that this change will have a much stronger runtime impact, since getting the answer from a widget can sometimes trigger deferred computation that results in other behavior. I think what you are doing here is the best way to do this check, though, so now that I see how we'd have to implement it, I think we should go back to making the save each time.

Sorry for the extra work.

Reply:
More of an FYI that this revert seems to be resulting in https://dimagi-dev.atlassian.net/browse/SUPPORT-18064?focusedCommentId=298192 when the image question is in a question list and we probably need to make a change here. @avazirna planning to debug more and explore this further.